include/alp/algorithms/cholesky.hpp

```diff
@@ -157,9 +165,364 @@ namespace alp {
 #endif
 			}
 
-			assert( rc == SUCCESS );
 			return rc;
 		}
 
+		/**
+		 * @brief Computes the blocked version Cholesky decomposition LL^H = H of a real symmetric
+		 *        or complex hermitian positive definite (SPD) matrix H where \a L is lower triangular.
+		 *        L^H  is equvalent to cojugate(transpose(L))
+		 *
+		 * @tparam D        Data element type
+		 * @tparam Ring     Type of the semiring used in the computation
+		 * @tparam Minus    Type minus operator used in the computation
+		 * @tparam Divide   Type of divide operator used in the computation
+		 * @param[out] L    output upper triangular matrix
+		 * @param[in]  H    input real symmetric (or complex hermitian) positive definite matrix
+		 * @param[in]  ring The semiring used in the computation
+		 * @return RC        SUCCESS if the execution was correct
+		 *
+		 */
+		template<
+			typename MatL,
+			typename MatH,
+			typename D = typename MatL::value_type,
+			typename Ring = Semiring< operators::add< D >, operators::mul< D >, identities::zero, identities::one >,
+			typename Minus = operators::subtract< D >,
+			typename Divide = operators::divide< D >,
+			std::enable_if_t<
+				is_matrix< MatL >::value &&
+				is_matrix< MatH >::value &&
+				structures::is_a< typename  MatL::structure, structures::UpperTriangular >::value &&
+				structures::is_a< typename  MatH::structure, structures::Symmetric >::value &&
+				is_semiring< Ring >::value &&
+				is_operator< Minus >::value &&
+				is_operator< Divide >::value
+			> * = nullptr
+		>
+		RC cholesky_uptr_blk(
+			MatL &L,
+			const MatH &H,
+			const size_t &bs,
+			const Ring &ring = Ring(),
+			const Minus &minus = Minus(),
+			const Divide &divide = Divide()
+		) {
+			(void) divide;
+			const Scalar< D > zero( ring.template getZero< D >() );
+
+			if (
+				( nrows( L ) != nrows( H ) ) ||
+				( ncols( L ) != ncols( H ) )
+			) {
+				std::cerr << "Incompatible sizes in cholesky_uptr_blk.\n";
+				return FAILED;
+			}
+
+			RC rc = SUCCESS;
+
+			const size_t n = nrows( L );
+
+			Matrix< D, typename MatH::structure > LL( n );
+			rc = rc ? rc : set( LL, H );
+#ifdef DEBUG
+			if( rc != SUCCESS ) {
+				std::cout << "set failed\n";
+				return rc;
+			}
+#endif
+
+			//nb: number of blocks of (max) size bz
+			size_t nb = n / bs ;
+			if( n % bs != 0 ){
+				nb = nb + 1 ;
+			}
+
+
+			for( size_t i = 0; i < nb ; ++i ) {
+				size_t a = std::min( i * bs, n );
+				size_t b = std::min( ( i + 1 ) * bs, n );
+				size_t c = n;
+
+				auto range1 = utils::range( a, b );
```

Comment:
better to write like this: utils::range range1( a, b )
Maybe even declare as const